pydbus/xml_generator.py

```diff
@@ -0,0 +1,107 @@
+"""Automatic XML documentation generator."""
+import inspect
+import sys
+
+from xml.etree import ElementTree
+
+
+PROPERTY_EMITS_SIGNAL = "org.freedesktop.DBus.Property.EmitsChangedSignal"
+
+
+# Python 2 treats them as methods, Python 3 as functions
+ismethod = inspect.ismethod if sys.version_info[0] == 2 else inspect.isfunction
+
+
+def verify_arguments(function):
+	"""Verify that the function is correctly defined."""
+	args, vargs, kwargs, defaults = inspect.getargspec(function)
+	if defaults is not None:
+		raise ValueError(
+			"Default values are not allowed for method "
+			"'{}'".format(function.__name__))
+	# TODO: Check if vargs or kwargs are actually allowed in dbus.
```

Comment:
Ah okay thanks for the information (both that vargs and kwargs are not allowed and that you have `dbus_context`). But do I see it correctly that this `dbus_context` has nothing to do with `vargs` or `kwargs`? If a method has keyword arguments it won't populate `dbus_context` would it? (But nevertheless it should skip those arguments)